`luarocks install tl` fails on Windows

Windows 11, MSVC 
Running this command within the MSVC developer command prompt, got the following error:

```
C:\Users\X>luarocks install tl
Installing https://luarocks.org/tl-0.24.3-1.src.rock
Using dependencies pinned in lockfile: luarocks.lock
tl 0.24.3-1 is pinned to argparse 0.7.1-1 (0.7.1-1 installed: success)
tl 0.24.3-1 is pinned to compat53 0.12-1 (not installed)
Installing https://luarocks.org/compat53-0.12-1.src.rock
cl /nologo /MD /O2 -c -Folutf8lib.obj -IC:\Lua\lua-5.4.2_Win64_dll17_lib\include lutf8lib.c
lutf8lib.c
link -dll -def:utf8.def -out:C:\Users\X\AppData\Local\Temp\luarocks_build-compat53-0.12-1-668774\compat53\utf8.dll C:\Lua\lua-5.4.2_Win64_dll17_lib\lua54.lib lutf8lib.obj
Microsoft (R) Incremental Linker Version 14.42.34435.0
Copyright (C) Microsoft Corporation.  All rights reserved.

   Creating library C:\Users\X\AppData\Local\Temp\luarocks_build-compat53-0.12-1-668774\compat53\utf8.lib and object C:\Users\X\AppData\Local\Temp\luarocks_build-compat53-0.12-1-668774\compat53\utf8.exp
cl /nologo /MD /O2 -c -Folstrlib.obj -IC:\Lua\lua-5.4.2_Win64_dll17_lib\include lstrlib.c
lstrlib.c
link -dll -def:string.def -out:C:\Users\X\AppData\Local\Temp\luarocks_build-compat53-0.12-1-668774\compat53\string.dll C:\Lua\lua-5.4.2_Win64_dll17_lib\lua54.lib lstrlib.obj
Microsoft (R) Incremental Linker Version 14.42.34435.0
Copyright (C) Microsoft Corporation.  All rights reserved.

   Creating library C:\Users\X\AppData\Local\Temp\luarocks_build-compat53-0.12-1-668774\compat53\string.lib and object C:\Users\X\AppData\Local\Temp\luarocks_build-compat53-0.12-1-668774\compat53\string.exp
cl /nologo /MD /O2 -c -Foltablib.obj -IC:\Lua\lua-5.4.2_Win64_dll17_lib\include ltablib.c
ltablib.c
link -dll -def:table.def -out:C:\Users\X\AppData\Local\Temp\luarocks_build-compat53-0.12-1-668774\compat53\table.dll C:\Lua\lua-5.4.2_Win64_dll17_lib\lua54.lib ltablib.obj
Microsoft (R) Incremental Linker Version 14.42.34435.0
Copyright (C) Microsoft Corporation.  All rights reserved.

   Creating library C:\Users\X\AppData\Local\Temp\luarocks_build-compat53-0.12-1-668774\compat53\table.lib and object C:\Users\X\AppData\Local\Temp\luarocks_build-compat53-0.12-1-668774\compat53\table.exp
cl /nologo /MD /O2 -c -Foliolib.obj -IC:\Lua\lua-5.4.2_Win64_dll17_lib\include liolib.c
liolib.c
liolib.c(280): warning C4047: '=': 'FILE *' differs in levels of indirection from 'int'
link -dll -def:io.def -out:C:\Users\X\AppData\Local\Temp\luarocks_build-compat53-0.12-1-668774\compat53\io.dll C:\Lua\lua-5.4.2_Win64_dll17_lib\lua54.lib liolib.obj
Microsoft (R) Incremental Linker Version 14.42.34435.0
Copyright (C) Microsoft Corporation.  All rights reserved.

io.def : error LNK2001: unresolved external symbol luaopen_compat53_io
C:\Users\X\AppData\Local\Temp\luarocks_build-compat53-0.12-1-668774\compat53\io.lib : fatal error LNK1120: 1 unresolved externals

Error: Failed installing dependency: https://luarocks.org/compat53-0.12-1.src.rock - Build error: Failed compiling module compat53\io.dll

```

`luarocks list` gives the following output (just fyi):

```
C:\Users\X>luarocks list

Rocks installed for Lua 5.4
---------------------------

argparse
   0.7.1-1 (installed) - C:\Users\X\AppData\Roaming\luarocks\lib\luarocks\rocks-5.4

compat53
   0.14.3-1 (installed) - C:\Users\X\AppData\Roaming\luarocks\lib\luarocks\rocks-5.4

luasocket
   3.1.0-1 (installed) - C:\Users\X\AppData\Roaming\luarocks\lib\luarocks\rocks-5.4



```

I did a clean `luarocks install tl`, and the same linker error still occured.

```
C:\Users\Lin>luarocks install tl
Installing https://luarocks.org/tl-0.24.3-1.src.rock
Using dependencies pinned in lockfile: luarocks.lock
tl 0.24.3-1 is pinned to argparse 0.7.1-1 (not installed)
Installing https://luarocks.org/argparse-0.7.1-1.src.rock
No existing manifest. Attempting to rebuild...
argparse 0.7.1-1 is now installed in C:\Users\Lin\AppData\Roaming\luarocks (license: MIT)

tl 0.24.3-1 is pinned to compat53 0.12-1 (not installed)
Installing https://luarocks.org/compat53-0.12-1.src.rock
cl /nologo /MD /O2 -c -Folutf8lib.obj -IC:\Lua\lua-5.4.2_Win64_dll17_lib\include lutf8lib.c
lutf8lib.c
link -dll -def:utf8.def -out:C:\Users\Lin\AppData\Local\Temp\luarocks_build-compat53-0.12-1-4239533\compat53\utf8.dll C:\Lua\lua-5.4.2_Win64_dll17_lib\lua54.lib lutf8lib.obj
Microsoft (R) Incremental Linker Version 14.42.34435.0
Copyright (C) Microsoft Corporation.  All rights reserved.

   Creating library C:\Users\Lin\AppData\Local\Temp\luarocks_build-compat53-0.12-1-4239533\compat53\utf8.lib and object C:\Users\Lin\AppData\Local\Temp\luarocks_build-compat53-0.12-1-4239533\compat53\utf8.exp
cl /nologo /MD /O2 -c -Folstrlib.obj -IC:\Lua\lua-5.4.2_Win64_dll17_lib\include lstrlib.c
lstrlib.c
link -dll -def:string.def -out:C:\Users\Lin\AppData\Local\Temp\luarocks_build-compat53-0.12-1-4239533\compat53\string.dll C:\Lua\lua-5.4.2_Win64_dll17_lib\lua54.lib lstrlib.obj
Microsoft (R) Incremental Linker Version 14.42.34435.0
Copyright (C) Microsoft Corporation.  All rights reserved.

   Creating library C:\Users\Lin\AppData\Local\Temp\luarocks_build-compat53-0.12-1-4239533\compat53\string.lib and object C:\Users\Lin\AppData\Local\Temp\luarocks_build-compat53-0.12-1-4239533\compat53\string.exp
cl /nologo /MD /O2 -c -Foliolib.obj -IC:\Lua\lua-5.4.2_Win64_dll17_lib\include liolib.c
liolib.c
liolib.c(280): warning C4047: '=': 'FILE *' differs in levels of indirection from 'int'
link -dll -def:io.def -out:C:\Users\Lin\AppData\Local\Temp\luarocks_build-compat53-0.12-1-4239533\compat53\io.dll C:\Lua\lua-5.4.2_Win64_dll17_lib\lua54.lib liolib.obj
Microsoft (R) Incremental Linker Version 14.42.34435.0
Copyright (C) Microsoft Corporation.  All rights reserved.

io.def : error LNK2001: unresolved external symbol luaopen_compat53_io
C:\Users\Lin\AppData\Local\Temp\luarocks_build-compat53-0.12-1-4239533\compat53\io.lib : fatal error LNK1120: 1 unresolved externals

Error: Failed installing dependency: https://luarocks.org/compat53-0.12-1.src.rock - Build error: Failed compiling module compat53\io.dll

C:\Users\Lin>luarocks list

Rocks installed for Lua 5.4
---------------------------

argparse
   0.7.1-1 (installed) - C:\Users\Lin\AppData\Roaming\luarocks\lib\luarocks\rocks-5.4


```

--- a/luarocks.lock
+++ b/luarocks.lock
@@ -1,7 +1,7 @@
 return {
    dependencies = {
       argparse = "0.7.1-1",
-      compat53 = "0.12-1",
+      compat53 = "0.14.4-1",
       luafilesystem = "1.8.0-1"
    },
    test_dependencies = {

--- a/spec/lang/parser/parser_error_spec.lua
+++ b/spec/lang/parser/parser_error_spec.lua
@@ -1,4 +1,5 @@
 local tl = require("tl")
+local util = require("spec.util")
 
 describe("parser errors", function()
    it("parse errors include filename", function ()
@@ -7,25 +8,11 @@ describe("parser errors", function()
    end)
 
    it("parse errors in a required package include filename of required file", function ()
-      local io_open = io.open
-      finally(function() io.open = io_open end)
-      io.open = function (filename, mode)
-         if string.match(filename, "bar.tl$") then
-            -- Return a stub file handle
-            return {
-               read = function (_, format)
-                  if format == "*a" then
-                     return "local x 1"                  -- Return fake bar.tl content
-                  else
-                     error("Not implemented!")  -- Implement other modes if needed
-                  end
-               end,
-               close = function () end,
-            }
-         else
-            return io_open(filename, mode)
-         end
-      end
+      util.mock_io(finally, {
+         ["bar.tl"] = [[
+            local x 1
+         ]],
+      })
 
       local code = [[
          local bar = require "bar"

--- a/spec/util.lua
+++ b/spec/util.lua
@@ -96,6 +96,8 @@ function util.mock_io(finally, filemap)
    assert(type(finally) == "function")
    assert(type(filemap) == "table")
 
+   local io = package.loaded["compat53.module"] and require("compat53.module").io or io
+
    local io_open = io.open
    on_finally(finally, function() io.open = io_open end)
    io.open = function (filename, mode)

